janitor/utils.py

```diff
@@ -2373,3 +2376,719 @@ def _sub_process_text_result_MultiIndex(index: pd.MultiIndex, result, df):
     # (# extra_index_line)
     df = df.droplevel(-1).set_index("match", append=True)
     return df
+
+
+def _check_operator(op: str):
+    """
+    Check that operator is one of
+    `>`, `>=`, `==`, `!=`, `<`, `<=`.
+    Used in `conditional_join`.
+    """
+    if op not in ("<", ">", "<=", ">=", "==", "!="):
```

Comment:
Tuples can be changed to Set if all we need to do is check for membership. All the Operators can be changed to Enums and used

Reply:
probably should update `check` function in `utils` to use sets. NIce pickup @nvamsikrishna05